### PR TITLE
Added Documentation for Expand and Collapse UG

### PR DESCRIPTION
**Pull request by SuthiYuvaraj**
Added Documentation for Expand and Collapse UG

### CODE REVIEW

File: MAUI/TreeView/expand-collapse.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+documentation: ug
+---
+
+# Expand and Collapse in Xamarin TreeView (SfTreeView)
+
+The TreeView allows you to expand and collapse the nodes either by user interaction on the nodes or by programmatically. 
+
+##  Expand Action Target
+
+ Expanding and Collapsing of nodes can be performed either by tapping the expander view or in both expander view and content view by setting the `ExpandActionTarget` property.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:16]: Expanding and collapsing of nodes can be performed by tapping the expander view or by setting the `ExpandActionTarget` property in both the expander view and content view.

File: MAUI/TreeView/expand-collapse.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+{% tabs %}
+{% highlight xaml %}
+
+<syncfusion:SfTreeView x:Name="TreeView" ExpandActionTarget="Node"/>
+
+{% endhighlight %}
+{% highlight c# %}
+
+// Expands by tapping both expander view and content view.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:26]: // Extends by tapping both the expander and content views.

File: MAUI/TreeView/expand-collapse.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+treeView.ExpandActionTarget = Syncfusion.Maui.TreeView.TreeViewExpandActionTarget.Node;
+
+{% endhighlight %}
+{% endtabs %}
+
+## Auto Expand Mode
+
+By default, the treeview items will be in collapsed state. You can define how the nodes to be expanded while loading the TreeView by using `AutoExpandMode` property.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:34]: By default, the items in the treeview will be in a collapsed state. You can define how the nodes should be expanded when loading the TreeView by using the `AutoExpandMode` property.

File: MAUI/TreeView/expand-collapse.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+The `AutoExpandMode` property is only applicable for bound mode. For Unbound mode you need to set `IsExpanded` property to `true` while creating the nodes, to be in expanded state while loading the TreeView.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:36]: The `AutoExpandMode` property is only applicable in bound mode. In the unbound mode, you need to set the `IsExpanded` property to `true` when creating the nodes in order for them to be in the expanded state when the TreeView is loaded.

File: MAUI/TreeView/expand-collapse.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+* None : All items are collapsed when loaded.
+* RootNodesExpanded : Expands only the root item when loaded.
+* AllNodesExpanded : Expands all the items when loaded.
+
+## Programmatic Expand and Collapse
+
+TreeView allows programmatic expand and collapse based on the `TreeViewNode` and level by using following methods.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:44]: TreeView allows programmatic expansion and collapse based on the TreeViewNode and level using the following methods.

File: MAUI/TreeView/expand-collapse.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+* `ExpandNode(TreeViewNode item)` - Method to expand the particular `TreeViewNode` passed to it.
+* `CollapseNode(TreeViewNode item)` - Method to collapse the particular `TreeViewNode` passed to it.
+* `ExpandNodes(int level)` - Method to expand the all items of level passed to it.
+* `CollapseNodes(int level)` - Method to expand the all items of level passed to it.
+
+{% tabs %}
+{% highlight c# %}
+
+// Expands all the nodes of root level '0'
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:54]: // Expands all the nodes at root level '0'.

File: MAUI/TreeView/expand-collapse.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+treeView.ExpandNodes(0);
+
+// Collapses all the nodes of root level '0'
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:57]: // Collapses all the nodes at root level '0'.

File: MAUI/TreeView/expand-collapse.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+// Expand a particular node.
+treeView.CollapseNode(node);
+
+{% endhighlight %}
+{% endtabs %}
+
+### Expand and Collapse all the nodes
+
+You can expand and collapse all the `TreeViewNode` programmatically at runtime by using the `SfTreeView.ExpandAll` method and `SfTreeView.CollapseAll` method.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:71]: Programmatically expand and collapse all the `TreeViewNode` at runtime using the `SfTreeView.ExpandAll` and `SfTreeView.CollapseAll` methods.

File: MAUI/TreeView/expand-collapse.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+{% tabs %}
+{% highlight c# %}
+
+//Expands all the nodes
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:76]: //Expands all the nodes.

File: MAUI/TreeView/expand-collapse.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+treeView.ExpandAll();
+
+//Collapses all the nodes
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:79]: //Collapses all the nodes.

File: MAUI/TreeView/expand-collapse.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+treeView.CollapseAll();
+
+{% endhighlight %}
+{% endtabs %}
+
+## Expand and Collapse using Keyboard
+
+TreeView allows to expand and collapse the nodes by using right and left arrows keys. To expand a node, press the right arrow key and to collapse a node, press the left arrow key on the focused item.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:87]: TreeView allows the expansion and collapse of the nodes using the right and left arrow keys. To expand a node, press the right arrow key; to collapse a node, press the left arrow key on the focused item.

File: MAUI/TreeView/expand-collapse.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+## Events
+
+TreeView exposes following events to handle expanding and collapsing of items.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:91]: TreeView exposes the following events to handle the expanding and collapsing of items.

File: MAUI/TreeView/expand-collapse.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+* `NodeCollapsing` - It occurs when a node is being collapsed.
+* `NodeExpanding` - It occurs when a node is being expanded.
+* `NodeCollapsed` - It occurs when a node is collapsed.
+* `NodeExpanded`  - It occurs when a node is expanded.
+
+The expanding and collapsing interactions can be handled with the help of `NodeCollapsing` and `NodeExpanding` events and expanded and collapsed interactions can be handled with help of `NodeCollapsed` and `NodeExpanded` events.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:98]: The expanding and collapsing interactions can be handled with the help of the `NodeCollapsing` and `NodeExpanding` events, and the expanded and collapsed interactions can be handled with the help of the `NodeCollapsed` and `NodeExpanded` events.

File: MAUI/TreeView/expand-collapse.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+You can also achieve handle expand and collapse operation using `ExpandCommand` and `CollapseCommand`.
```
Comment (florenceliliana) [inline MAUI/TreeView/expand-collapse.md:100]: You can also achieve handling of expand and collapse operations using the `ExpandCommand` and `CollapseCommand`.

**Review by florenceliliana — COMMENTED** [review]
The content review is completed.

**Review by Lakshminatarajan — APPROVED** [review]
Approved

**Pull request merged**